### DISCUSSION TITLE
My feedback

### DISCUSSION THREAD

**Discussion by frog-o** [live-codes/livecodes#303] [General]
@hatemhosny , you ask for my opinion when i had the discussion from [Bundlejs ](https://github.com/okikio/bundlejs/discussions/45#discussioncomment-5285060)  (I have the email )
wich i am happy to give but people should know i am nobody and very critical(wich is probably the reason why i don't get any where and a nobody). I spend a lot of time on research just to find out somebody already done the work or it's unwanted feature. I trying to change that.

Is there a reason you aren't using jquery , react, vuejs or something like that IMHO it would make your code a lot more readable ?
I believe they all can be used after building Wich you are using .  Also something interesting 🤔 to me is bable-standalone.  

It is a terrific start but to be honest  I originally hated it at first glance.

here the bad things  
 - The load screen  is a movie
 while to most a must have they are hard on me and every time I see them I close they page and look else where.    
-  no pyscript option to start your own template from the main screen
-config all over the place
Here the good 
- good feature set 👍
- document was terrific for beta 🙂
- dos most of what i want 😀

I would like to know what import to you and what you would like help with if anything(not saying I can)
I looking forward to playing with the source on the weekends 

Here are what I would like to try to do
- add theme support
- Add a file manager
-  move the preview to tab on top
-  move the split screen to bottom when view with is narrower than top

I would love to hear your thoughts or anyone else on what libraries I should use or if you where working on anyone of these. I always thought that what discussion where for to get help from the community add plan together on implementing new features.

**Comment by hatemhosny** [comment]
hi @frog-o,

I'm sorry for the (very) late reply! I totally missed your post.

Thank you for having the time to review LiveCodes and send your comments. You have very valid points. I'll try to answer some of them here:

Regarding the use of a library/framework, of course, this app could have been built with one of the many currently available options. It was mainly a personal preference to use (vanilla) TypeScript with esbuild. When I first started the project, I did not expect it to grow that big. I still feel this gives me more low-level control. However, some better code organization can make it more readable. I will try to do that.

Fair comment about the loading animation. I'll consider using a more subtle one, specially for embedded playgrounds.

Making the UI simpler is a good point. Now, there is a "Welcome screen", which gives easier access to the most frequently used functionality. It is always difficult to make the balance between being powerful and simple.

Thank you for showing interest to contribute. I think adding custom themes can be a good start. 

If you plan to contribute, before actually starting, please open an issue so that we can discuss the approach.

Thank you very much, and sorry for missing your post.
